File: sdks/php/src/Types/PlanOverrideResponse.php
<?php

namespace NovaBilling\Types;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;
use DateTime;
use NovaBilling\Core\Types\Date;

class PlanOverrideResponse extends JsonSerializableType
{
    /**
     * @var string $id
     */
    #[JsonProperty('id')]
    public string $id;

    /**
     * @var string $customerId
     */
    #[JsonProperty('customerId')]
    public string $customerId;

    /**
     * @var string $planId
     */
    #[JsonProperty('planId')]
    public string $planId;

    /**
     * @var ?array<string, mixed> $overriddenPrices
     */
    #[JsonProperty('overriddenPrices'), ArrayType(['string' => 'mixed'])]
    public ?array $overriddenPrices;

    /**
     * @var ?float $overriddenMinimumCommitment
     */
    #[JsonProperty('overriddenMinimumCommitment')]
    public ?float $overriddenMinimumCommitment;

    /**
     * @var ?array<string, mixed> $overriddenCharges
     */
    #[JsonProperty('overriddenCharges'), ArrayType(['string' => 'mixed'])]
    public ?array $overriddenCharges;

    /**
     * @var ?array<string, mixed> $metadata
     */
    #[JsonProperty('metadata'), ArrayType(['string' => 'mixed'])]
    public ?array $metadata;

    /**
     * @var DateTime $createdAt
     */
    #[JsonProperty('createdAt'), Date(Date::TYPE_DATETIME)]
    public DateTime $createdAt;

    /**
     * @var DateTime $updatedAt
     */
    #[JsonProperty('updatedAt'), Date(Date::TYPE_DATETIME)]
    public DateTime $updatedAt;

    /**
     * @param array{
     *   id: string,
     *   customerId: string,
     *   planId: string,
     *   createdAt: DateTime,
     *   updatedAt: DateTime,
     *   overriddenPrices?: ?array<string, mixed>,
     *   overriddenMinimumCommitment?: ?float,
     *   overriddenCharges?: ?array<string, mixed>,
     *   metadata?: ?array<string, mixed>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->id = $values['id'];
        $this->customerId = $values['customerId'];
        $this->planId = $values['planId'];
        $this->overriddenPrices = $values['overriddenPrices'] ?? null;
        $this->overriddenMinimumCommitment = $values['overriddenMinimumCommitment'] ?? null;
        $this->overriddenCharges = $values['overriddenCharges'] ?? null;
        $this->metadata = $values['metadata'] ?? null;
        $this->createdAt = $values['createdAt'];
        $this->updatedAt = $values['updatedAt'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: sdks/rust/src/api/types/plan_override_response.rs
pub use crate::prelude::*;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct PlanOverrideResponse {
    pub id: String,
    #[serde(rename = "customerId")]
    pub customer_id: String,
    #[serde(rename = "planId")]
    pub plan_id: String,
    #[serde(rename = "overriddenPrices")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub overridden_prices: Option<HashMap<String, serde_json::Value>>,
    #[serde(rename = "overriddenMinimumCommitment")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub overridden_minimum_commitment: Option<f64>,
    #[serde(rename = "overriddenCharges")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub overridden_charges: Option<HashMap<String, serde_json::Value>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<HashMap<String, serde_json::Value>>,
    #[serde(rename = "createdAt")]
    #[serde(with = "crate::core::flexible_datetime::offset")]
    pub created_at: DateTime<FixedOffset>,
    #[serde(rename = "updatedAt")]
    #[serde(with = "crate::core::flexible_datetime::offset")]
    pub updated_at: DateTime<FixedOffset>,
}